nativelink-config/src/stores.rs

```diff
@@ -441,19 +435,19 @@ pub enum StoreSpec {
     /// **Example JSON Config:**
     /// ```json
     /// "size_partitioning": {
-    ///     "size": 134217728, // 128mib.
-    ///     "lower_store": {
-    ///       "memory": {
-    ///         "eviction_policy": {
-    ///           "max_bytes": "${NATIVELINK_CAS_MEMORY_CONTENT_LIMIT:-100000000}"
-    ///         }
+    ///   "size": "128mib",
+    ///   "lower_store": {
+    ///     "memory": {
+    ///       "eviction_policy": {
+    ///         "max_bytes": "${NATIVELINK_CAS_MEMORY_CONTENT_LIMIT:-100000000}"
```

Comment:
I'm not sure whether its available to do this or not, can we just use like `${NATIVELINK_CAS_MEMORY_CONTENT_LIMIT:-100000000}` to `${NATIVELINK_CAS_MEMORY_CONTENT_LIMIT:-100mb}`?

Reply:
another good comment! @palfrey wdyt about supporting this format? There are risks but I think we need to weigh flexibility, consistency, and usability in the config. mixing math and letters is tough sometimes.

Reply:
We already support it :) It's done via [shellexpand](https://docs.rs/shellexpand/latest/shellexpand/) which happily supports this as an option. I've changed the example config to show this.